patches/server/0738-Fix-entities-glitching-through-blocks-pushed-by-pist.patch

```diff
@@ -0,0 +1,80 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Peter Svensson <petersv@psv.nu>
+Date: Mon, 2 Aug 2021 12:06:05 +0200
+Subject: [PATCH] Fix entities glitching through blocks pushed by pistons
+ (#5189)
+
+Fix #5189 where entities in slime blocks are unreliable.
+
+Add a general mechanism where the entity knows how many ticks into
+the future that it has to be fully ticked. ActivationRange respects
+that and does not cull any of those ticks to the entity.
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 896d892237b29eb404398db07264eb6f04786754..7981965bfb554709d3b25eb8ae98f3e6853b9e64 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -690,6 +690,11 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+ 
+         this.firstTick = false;
+         this.level.getProfiler().pop();
++        // Paper start - track when all ticks are required for the entity, e.g. after a piston push
++        if (this.isTickRequired()) {
++           --this.remainingExactTicks;
++        }
++        // Paper end
+     }
+ 
+     public void setSharedFlagOnFire(boolean onFire) {
+@@ -3991,5 +3996,16 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, n
+     public boolean isTicking() {
+         return ((ServerChunkCache) level.getChunkSource()).isPositionTicking(this);
+     }
++ 
++    // Track when all ticks are required for the entity, e.g. after a piston push
++    private int remainingExactTicks = 0;
```

Comment:
Oki. I was asked to make it private (was origianlly public) and introduce getters/setters. Are you sure you want this to be direct access instead of setter/getter?

Reply:
Yes, to make it more in-line with the original EAR patch.